src/lib/libast/cdt/dtrehash.c

```diff
@@ -648,7 +648,7 @@ static int hashevent(Dt_t *dt, int event, Void_t *arg) {
     Dtdisc_t *disc = dt->disc;
     Hash_t *hash = (Hash_t *)dt->data;
 
-    if (!(disc = dt->disc)) return -1;
+    if (!disc) return -1;
```

Comment:
I fixed several instances of this pattern when I did the mass restyling of the code. /me is a sad puppy that there are so many silly mistakes of this sort in the code. All anyone had to do was pay attention to what the compiler was reporting.